variables.tf

```diff
@@ -99,105 +70,172 @@ variable "k6_ip_ranges_url" {
   type        = string
 }
 
-variable "country_rates" {
+## Variables for WAF Rules
+
+variable "whitelisted_ips_v4" {
   default     = []
-  description = "Countries blocking limits"
-  type = list(object({
-    name         = string
-    limit        = number
-    priority     = number
-    country_code = set(string)
-  }))
-  # Example
-  # [
-  #   { name         = "Group_1-CH"
-  #     limit        = 50000
-  #     country_code = ["CH"]
-  #     priority     = 20
-  #   },
-  #   { name         = "Group_2-DE_AT_FR"
-  #     limit        = 4000
-  #     country_code = ["AT", "FR", "DE"]
-  #     priority     = 21
-  #   },
-  #   ...
-  #   { name         = "Very_slow"
-  #     limit        = 100
-  #     country_code = ["AR", "BD", "BR", "KH", "CN", "CO", "EC", "IN", "ID", "MX", "NP", "PK", "RU", "SG", "TR", "UA", "AE", "ZM", "VN"]
-  #     priority     = 25
-  #   }
-  # ]
+  description = "List of enterprise IP ranges to be whitelisted. Set to empty list to disable the whitelisting"
+  type        = list(string)
+  validation {
+    condition = alltrue([
+      for ip in var.whitelisted_ips_v4 : can(regex("^\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}/\\d{1,2}", ip))
+    ])
+    error_message = "whitelisted_ips_v4 must contain valid IP V4 ranges with mask. Example: ['1.1.1.1/16', '255.255.255.255/32']"
+  }
 }
 
-variable "everybody_else_limit" {
-  default     = 0
-  description = "The blocking limit for all countries which are not covered by country_rates - not applied if it set to 0"
-  type        = number
+variable "whitelisted_ips_v6" {
+  default     = []
+  description = "The IPv6 to allow"
+  type        = list(string)
+  validation {
+    # Not the "real" regexp for ipv6. The right one has around 1000 characters...
+    condition = alltrue([
+      for ip in var.whitelisted_ips_v6 : can(regex("^[0-9a-fA-F:]*/\\d{1,3}", ip))
+    ])
+    error_message = "whitelisted_ips_v6 must contain valid IP V6 ranges."
+  }
 }
 
-variable "aws_managed_rules" {
+variable "whitelisted_hostnames" {
   default     = []
-  description = "AWS managed rules for WAF to set. Not applicable for var.waf_scope = REGIONAL"
+  description = "Whitelisted host headers. Example: ['partner-xxxxx.yyyyy.domain.ch']"
```

Comment:
@DCamma Does this mean we can whitelist all the request having a Host header? Are we using such a rule in our waf to whitelist others? It can be overcome super easily since I can craft requests with the "Host" header as I want - [link](https://everything.curl.dev/http/modify/headers) (I know it's not part of your work)

Reply:
True, we are using it in Disco and I wanted to keep the functionality having it more general: https://github.com/DND-IT/disco-infra-terraform/blob/9bba656931b1f57d9d3ed1d8f1f173e98f9037d6/cloudfront.tf#L287
But maybe we should remove the functionality if It's dangerous

Reply:
Since it will be a breaking change to remove the functionality, I'd vote to understand if it's still necessary considering the risk of keeping an open door to the DDOS'ers. I'll ask in the #it_pts_dai channel

Reply:
removed